src/App.js

```diff
@@ -289,9 +289,11 @@ class App extends Component {
       return;
     }
     this.budgetDeduction(index);
-    this.setState({
-      notReturned: Number((notReturned - sharedAmount).toFixed(2))
-    });
+    if (transaction[index].toggle) {
+      this.setState({
+        notReturned: Number((notReturned - sharedAmount).toFixed(2))
```

Comment:
To już trochę zbyt abstrakcyjne co wkładasz do tego setState. Wywal to do zmiennej, która w jakiś sposób opisuje to co tutaj robisz.